src/components/main-grid/grid-item/index.js

```diff
@@ -0,0 +1,34 @@
+// @vendors
+import { object, string } from "prop-types";
+
+// @styles
+import styles from "./styles.module.scss";
+
+const GridITem = ({ data }) => {
+  const { name, category, price, mainimage: mainImg } = data;
+  const imgStyle = {
+    backgroundImage: `url(${mainImg.url})`,
+    height: mainImg.height,
+    width: mainImg.width,
+  };
+
+  return (
+    <div key={category.id} className={styles.item} style={imgStyle}>
```

Comment:
nit: It would be better for optimization and seo to use <img> instead of backgroundImage if possible